File: src/CENTRO_SALUD/Profesional.java
package CENTRO_SALUD;

import java.util.ArrayList;
import java.util.List;

public class Profesional extends Persona {
    private String especialidad;
    private String cargo;
    private List<Cita> agenda = new ArrayList<>(); // 1 profesional -> N citas
 
    public Profesional(String idPersona, String dni, String nombres, String apellidos,
                        String especialidad, String cargo) {
        super(idPersona, dni, nombres, apellidos);
        this.especialidad = especialidad;
        this.cargo = cargo;
    }

    public void agregarCita(Cita cita) {
        agenda.add(cita);
    }

    public List<Cita> getAgenda() { return agenda; }
 
    public String getEspecialidad() { return especialidad; }
    public void setEspecialidad(String especialidad) { this.especialidad = especialidad; }
 
    public String getCargo() { return cargo; }
    public void setCargo(String cargo) { this.cargo = cargo; }
}

File: centro_salud_python/profesional.py
from persona import Persona


class Profesional(Persona):
    """Un profesional de salud (medico, enfermero, etc)."""

    def __init__(self, id_persona, dni, nombres, apellidos, especialidad, cargo):
        super().__init__(id_persona, dni, nombres, apellidos)
        self._especialidad = especialidad
        self._cargo = cargo
        self._agenda = []  # 1 profesional -> N citas

    def agregar_cita(self, cita):
        self._agenda.append(cita)

    def get_agenda(self):
        return self._agenda

    def get_especialidad(self):
        return self._especialidad

    def set_especialidad(self, especialidad):
        self._especialidad = especialidad

    def get_cargo(self):
        return self._cargo

    def set_cargo(self, cargo):
        self._cargo = cargo
